pennylane/devices/default_qubit.py

```diff
@@ -411,32 +430,19 @@ def supports_derivatives(
         """
         if execution_config is None:
             return True
-        # backpropagation currently supported for all supported circuits
-        # will later need to add logic if backprop requested with finite shots
-        # do once device accepts finite shots
-        if (
-            execution_config.gradient_method == "backprop"
-            and execution_config.device_options.get("max_workers", self._max_workers) is None
-            and execution_config.interface is not None
-        ):
-            return True
 
         if (
-            execution_config.gradient_method == "adjoint"
-            and execution_config.use_device_gradient is not False
+            execution_config.gradient_method in {"backprop", "best"}
+            and execution_config.device_options.get("max_workers", self._max_workers) is None
         ):
             if circuit is None:
                 return True
+            return not circuit.shots and not any(
+                isinstance(m.obs, qml.SparseHamiltonian) for m in circuit.measurements
+            )
 
-            prog = TransformProgram()
-            _add_adjoint_transforms(prog)
-
-            try:
-                prog((circuit,))
-            except (qml.operation.DecompositionUndefinedError, qml.DeviceError):
-                return False
-            return True
-
+        if execution_config.gradient_method == "adjoint":
```

Comment:
I was under the impression that if backprop is not possible but adjoint is possible and `diff_method=="best"`, then we switch to adjoint.

Reply:
I was about to write that there isn't any case where adjoint isn't supported but backprop isn't, until I realized that this is the case for `max_workers is not None`.  I'll write in that logic so that we interpret `"best"` as `"adjoint"` if `max_workers is not None`.

Reply:
Does this still need to be updated?

Reply:
Oh thanks for the reminder. I'll do so right now.

Reply:
Now if `max_workers is not None`, we will fall back to using adjoint instead of backprop.